packages/katello/katello/hostname-change.rb

```diff
@@ -550,8 +550,12 @@ def run
         if success
           STDOUT.puts result
           STDOUT.puts 'Restarting puppet services'
-          run_cmd('/bin/systemctl try-restart puppet')
-          run_cmd('foreman-maintain service restart --only puppetserver')
+          puppet_services = []
+          ['puppet', 'puppetserver'].each do |service|
+            `systemctl is-active #{service} --quiet`
+            puppet_services << service if $?.success?
+          end
```

Comment:
Nit: this is the same, but shorter:
```suggestion
          puppet_services = ['puppet', 'puppetserver'].select do |service|
            system("systemctl is-active #{service} --quiet")
          end
```

Reply:
indeed.